build/bazel/remote/execution/v2/remote_execution.proto

```diff
@@ -1997,6 +1997,17 @@ message CacheCapabilities {
   // [BatchUpdateBlobs][build.bazel.remote.execution.v2.ContentAddressableStorage.BatchUpdateBlobs]
   // requests.
   repeated Compressor.Value supported_batch_update_compressors = 7;
+
+  // Maximum size a single CAS blob can have in the remote cache system.
+  // - If it is 0, it means there is no limit is set. Arbitrarily large blobs
+  //   may be uploaded to and downloaded from the cache.
+  // - If it is larger than 0, implementations SHOULD not attempt to upload
+  //   blobs with size larger than the limit. If they do, it is up to the
+  //   cache implementation what to do, e.g. accept the request anyway or reject
```

Comment:
I'm wondering if we should actually tighten this up to improve the possibility of interop between implementations. Being permissive sounds nice in theory, but if you're using a system that advertises limit X (but actually accepts larger blobs), and move to another system that advertises the same limit (and enforces it), you'd get surprising failures.

(Note that the carve-out below for *serving* larger blobs is still important, for dealing with legacy conditions.)